components/server/src/prebuilds/bitbucket-server-app.ts

```diff
@@ -153,7 +153,7 @@ export class BitbucketServerApp {
                 { span },
                 {
                     user: projectAndOwner.user,
-                    project: projectAndOwner?.project,
+                    project: projectAndOwner.project!,
```

Comment:
What's the reason we need the ! here?

Reply:
I changed the signature of `PrebuildManager.startPrebuild` and made the `Project` arg mandatory instead of throwing an error. 
In fact `PrebuildManager.shouldPrebuild` would return false if no project is present. 

As it's raising questions here, I'll rewrite and pull out the check from `PrebuildManager.shouldPrebuild`.

Reply:
That change made sense but it was also quite a rabbit hole: https://github.com/gitpod-io/gitpod/pull/18698/commits/be8fbc60e90a0aaf5e9aae2d5820e9539a1e384e

No one wants to review this. I'll need to think how to cut this PR in pieces.